modules/storage-api/src/main/java/org/apache/ignite/internal/storage/index/IndexRow.java

```diff
@@ -17,26 +17,22 @@
 
 package org.apache.ignite.internal.storage.index;
 
-import org.apache.ignite.internal.storage.PartitionStorage;
-import org.apache.ignite.internal.storage.SearchRow;
-
 /**
  * Represents an Index Row - a set of indexed columns and Primary Key columns (for key uniqueness).
  */
-public interface IndexRow {
+public interface IndexRow extends IndexBinaryRow {
     /**
-     * Returns the serialized presentation of this row as a byte array.
+     * Returns value of the indexed columns specified by index column order.
      *
-     * @return Serialized byte array value.
+     * @param idxColOrder Index column order.
+     * @return Value of the indexed columns.
      */
-    byte[] rowBytes();
+    Object value(int idxColOrder);
```

Comment:
May we need methods for primitive types to avoid boxing?

Reply:
I see nothing case where these methods may be used.